src/Parallel/Pin.cpp

```diff
@@ -65,28 +81,19 @@ cpu_set_t seissol::parallel::getWorkerUnionMask() {
   return workerUnion;
 }
 
-cpu_set_t seissol::parallel::getFreeCPUsMask() {
-  cpu_set_t workerUnion = getWorkerUnionMask();
-  cpu_set_t set;
-  CPU_ZERO(&set);
-  for (int i = 0; i < get_nprocs() ; ++i) {
-    CPU_SET(i, &set);
-  }
-  CPU_XOR(&set, &set, &workerUnion);
-
-  return set;
+cpu_set_t seissol::parallel::Pinning::getFreeCPUsMask() const {
+  return pthreadMask;
```

Comment:
(You can ignore the following comment if you want.)

Storing pthreadMask is actually redundant as you already store processMask and openmpMask. The operation to compute pthreadMask is really cheap.